Mu2eG4/src/Mu2eG4Inputs.cc

```diff
@@ -88,6 +89,44 @@ namespace mu2e {
 
     }
       break; // SimParticles
+
+    case Mu2eG4PrimaryType::StageParticles: {
+      std::optional<art::ProductID> simid;
+
+      auto const h = evt.getValidHandle<StageParticleCollection>(primaryTag_);
+      for(const auto& s : *h) {
+        if(!simid) {
+          simid = s.parent().id();
+
+          // We have to read the pointed-to product into memory by hand
+          // for the get(ProductID, &handle) call below to work.
+          // See Kyle's e-mail on the art-users list on 2021-03-10.
+          *s.parent();
+        }
+        else {
+          if(*simid != s.parent().id()) {
```

Comment:
I am missing the intent here: if simid was set in an earlier loop step, why require it to have the same parent?

Reply:
It is technically possible for different SimParticle ptrs to point into different collections.   We do not support such inputs, the code is to detect those cases and break rather then producing wrong output.   This should never trigger with normal Mu2e files.